test/parallel/test-net-socket-local-address.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var net = require('net');
+
+var conns = 0;
+var clientLocalPorts = [];
+var serverRemotePorts = [];
+
+var server = net.createServer(function(socket) {
+  serverRemotePorts.push(socket.remotePort);
+  conns++;
+});
+
+var client = new net.Socket();
+
+server.on('close', function() {
+  assert.deepEqual(clientLocalPorts, serverRemotePorts,
+                   'client and server should agree on the ports used');
+  assert.equal(2, conns);
+});
+
+server.listen(0, testConnect);
```

Comment:
Odd, I figured using the server socket's reported address would have been the most portable. Changed it to use `common.PORT` and `common.localhostIPv4`.